src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -218,7 +218,73 @@ public virtual int Next(int maxValue)
             return (int)(Sample() * maxValue);
         }
 
-        /*=====================================Next=====================================
+        private long FullLong()
+        {
+            long result = 0;
+
+            for (int i = 0; i < 4; i++)
+            {
+                short part;
+                unchecked
+                {
+                    part = (short) InternalSample();
+                }
```

Comment:
This has something similar to the effect of a modulus operation, but I haven't done the math to tell whether it will evenly generate all values in the range (due to `InternalSample()` not returning `int.MaxValue).  Possibly suspect.